hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncNonMetaRegionLocator.java

```diff
@@ -618,4 +635,48 @@ int getNumberOfCachedRegionLocations(TableName tableName) {
     return tableCache.regionLocationCache.getAll().stream()
       .mapToInt(RegionLocations::numNonNullElements).sum();
   }
+
+  private TimerTask getInvalidateMetaCacheTask(long metaCacheInvalidateInterval) {
+    return new TimerTask() {
+      @Override
+      public void run(Timeout timeout) throws Exception {
+        invalidateTableCache();
+        AsyncConnectionImpl.RETRY_TIMER.newTimeout(this, metaCacheInvalidateInterval,
+          TimeUnit.MILLISECONDS);
+      }
+    };
+  }
+
+  private void invalidateTableCache() {
+    // Should not throw any exception from here, it will affect schedule a new specified TimerTask.
+    Set<TableName> cachedTables = cache.keySet();
+    if (!cachedTables.isEmpty()) {
+      AsyncAdmin admin = conn.getAdmin();
+      for (TableName tableName : cachedTables) {
```

Comment:
To reduce the pressure to HMaster, I suggest here we do the request one by one, i.e, send a new isTableDisabled request in the callback of the previous request.

And in the last callback, we schedule the new timer task.